### PR TITLE
Build image on push

### PR DESCRIPTION
**Pull request by endrjuskr**

### CODE REVIEW

**Comment by endrjuskr** [comment]
On push to master we build and publish docker and publish documentation. We reuse build documentations from the previous job.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -74,14 +74,50 @@ jobs:
...
+        with:
+          name: docs
+          path: docs/build/html
+      - name: Publish documentation
+        uses: s0/git-publish-subdir-action@v2.2.0
+        env:
+          REPO: self
+          BRANCH: gh-pages
+          FOLDER: docs/build/html
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (TheMP) [inline .github/workflows/workflow.yml:99]: DId you maybe figure out where this token is coming from and who does it belong to?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -74,14 +74,50 @@ jobs:
...
+    if: contains(github.ref, 'master')
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to gcloud registry
+        id: gcloud
+        uses: elgohr/gcloud-login-action@0.2
+        with:
+          account_key: ${{ secrets.GCLOUD_KEY }}
+      - name: Publish to Registry
+        uses: elgohr/Publish-Docker-Github-Action@2.14
```
Comment (TheMP) [inline .github/workflows/workflow.yml:113]: for the future, will this also work with dockerhub?

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -22,13 +22,13 @@ jobs:
         id: cache
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt') }}
```
Comment (TheMP) [inline .github/workflows/workflow.yml:25]: Pls align with https://github.com/unit8co/u8timeseries/pull/55/files

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (endrjuskr) [reply .github/workflows/workflow.yml:114]: yes

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (endrjuskr) [reply .github/workflows/workflow.yml:100]: It is coming from Github Actions

**Pull request merged**